server/controllers/auth.js

```diff
@@ -7,124 +7,124 @@ const List = require("../models/List");
-    const token = generateToken(user._id);
-    const secondsInWeek = 604800;
-
-    res.cookie("token", token, {
-      httpOnly: true,
-      maxAge: secondsInWeek * 1000
-    });
-
-    res.status(201).json({
-      success: {
-        user: {
-          id: user._id,
-          username: user.username,
-          email: user.email
-        }
-      }
-    });
-    const shoppingList = {
-      name: "Shopping",
-      image:
-        "https://team-granny-smith-s3.s3.ca-central-1.amazonaws.com/default-images/Shopping.jpg",
-      userId: user._id
-    };
-    const wishlist = {
-      name: "Wishlist",
-      image:
-        "https://team-granny-smith-s3.s3.ca-central-1.amazonaws.com/default-images/Wishlist.jpg",
-      userId: user._id
-    };
-    await List.create([shoppingList, wishlist]);
-  } else {
-    res.status(400);
-    throw new Error("Invalid user data");
-  }
+	const { username, email, password } = req.body;
+
+	const emailExists = await User.findOne({ email });
+
+	if (emailExists) {
+		res.status(400);
+		throw new Error("A user with that email already exists");
+	}
+
+	const usernameExists = await User.findOne({ username });
+
+	if (usernameExists) {
+		res.status(400);
+		throw new Error("A user with that username already exists");
+	}
+
+	const user = await User.create({
+		username,
+		email,
+		password,
+	});
+
+	if (user) {
+		const token = generateToken(user._id);
+		const secondsInWeek = 604800;
+
+		res.cookie("token", token, {
+			httpOnly: true,
+			maxAge: secondsInWeek * 1000,
+		});
+
+		res.status(201).json({
+			success: {
+				user: {
+					id: user._id,
+					username: user.username,
+					email: user.email,
+				},
+			},
+		});
+		const shoppingList = {
+			name: "Shopping",
+			image: "https://team-granny-smith-s3.s3.ca-central-1.amazonaws.com/default-images/Shopping.jpg",
+			userId: user._id,
+			isPrivate: false,
+		};
+		const wishlist = {
+			name: "Wishlist",
+			image: "https://team-granny-smith-s3.s3.ca-central-1.amazonaws.com/default-images/Wishlist.jpg",
+			userId: user._id,
+			isPrivate: false,
+		};
+		await List.create([shoppingList, wishlist]);
+	} else {
+		res.status(400);
+		throw new Error("Invalid user data");
+	}
 });
 
 // @route POST /auth/login
 // @desc Login user
 // @access Public
 exports.loginUser = asyncHandler(async (req, res, next) => {
-  const { email, password } = req.body;
-
-  const user = await User.findOne({ email });
-
-  if (user && (await user.matchPassword(password))) {
-    const token = generateToken(user._id);
-    const secondsInWeek = 604800;
-
-    res.cookie("token", token, {
-      httpOnly: true,
-      maxAge: secondsInWeek * 1000
-    });
-
-    res.status(200).json({
-      success: {
-        user: {
-          id: user._id,
-          username: user.username,
-          email: user.email
-        }
-      }
-    });
-  } else {
-    res.status(401);
-    throw new Error("Invalid email or password");
-  }
+	const { email, password } = req.body;
```

Comment:
I think your prettier settings aren't aligned with the project settings, can you check that?

Reply:
I'll check